packages/compiler-cli/src/ngtsc/typecheck/src/type_check_block.ts

```diff
@@ -1273,27 +1273,47 @@ class TcbSwitchOp extends TcbOp {
   }
 
   override execute(): null {
-    const clauses: ts.CaseOrDefaultClause[] = [];
-
-    for (const current of this.block.cases) {
-      // Our template switch statements don't fall through so we always have a break at the end.
-      const breakStatement = ts.factory.createBreakStatement();
-      const clauseScope = Scope.forNodes(this.tcb, this.scope, null, current.children, null);
+    const expression = tcbExpression(this.block.expression, this.tcb, this.scope);
+    const root = this.generateCase(0, expression, null);
+    root && this.scope.addStatement(root);
```

Comment:
Compiler uses strict-null equality
```suggestion
    if (root !== null) {
      this.scope.addStatement(root);
    }
```